cmd/utils.go

```diff
@@ -161,12 +161,12 @@ func MigrateDatabase(ctx context.Context, migrationProjectId string, targetProfi
 
 func migrateSchema(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile,
 	ioHelper *utils.IOStreams, conv *internal.Conv, dbURI string, adminClient *database.DatabaseAdminClient) error {
-		spA := spanneraccessor.SpannerAccessorImpl{}
 		adminClientImpl, err := spanneradmin.NewAdminClientImpl(ctx)
 		if err != nil {
 			return err
 		}
-		err = spA.CreateOrUpdateDatabase(ctx, adminClientImpl, dbURI, sourceProfile.Driver, conv, sourceProfile.Config.ConfigType)
+		spA := spanneraccessor.SpannerAccessorImpl{AdminClient: adminClientImpl}
+		err = spA.CreateOrUpdateDatabase(ctx, dbURI, sourceProfile.Driver, conv, sourceProfile.Config.ConfigType)
```

Comment:
I suggest that we add a createNewSpannerAccessor implementation which returns and accessor with all three values, adminClient, instanceClient and spannerClient set and call that

Reply:
Different portions of the code do not need to initialize all the accessors. Most of the code references only use the adminClient, with instanceClient being used sparingly...
`createNewSpannerAccessor` method will result in a bunch of `null` being passed in various portions of the code to call it; feels more clunky to me..

Reply:
Having the method doesn't necessitate its usage. Since spanner accessor is passed as an argument in many places and hence its used it becomes clunky in my opinion. Just a thought.

Reply:
minor optimization - it seems like the adminClient is not being used in these place - and in other parts of the code. May make sense to change the method signature to just access a context and return the Spanner Accessor.

Reply:
Thanks! Had a discussion with @asthamohta offline. Will incorporate this - @bharadwaj-aditya we have followed a convention in other parts of the code that I will use here now.